test/versioned/langchain/helpers/custom-tool.js

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const { Tool } = require('@langchain/core/tools')
+const { makeGetRequestAsync } = require('../../../lib/agent_helper')
+
+module.exports = class TestTool extends Tool {
+  static lc_name() {
+    return 'TestTool'
+  }
+
+  name = 'node-agent-test-tool'
+  description = 'A test tool for versioned tests'
+  key
+
+  constructor(params) {
+    super()
+    this.baseUrl = params.baseUrl ?? this.baseUrl
+  }
+
+  async _call(uri) {
+    const url = `${this.baseUrl}/${uri}`
+    const res = await makeGetRequestAsync(url)
```

Comment:
Do we really need to make an HTTP request? Can't our tool just do some sort of transformation and return it?

Reply:
I guess so. I was hoping to make http to demonstrate that the external http call was being captured but because nock is introduced here it doesn't record the external http call. I'll simplify this